# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

**Provisional lane: Lane 2 — Refresh / Content Opportunity Scoring.**

Of the four lanes, this is the one where the starter data already shows a clear, actionable
pattern (see the numbers below): a large share of pages are trending down and a meaningful chunk
sit in "striking distance" of page one. That combination — visible, valuable, and slipping or
almost-there — is exactly the kind of messy, many-signal pattern that a hand-written rule
struggles to rank well, but a scored/ranked model can prioritize sensibly. It also maps cleanly
onto a real editorial workflow (a content team has limited hours and needs a queue, not just a
report), which makes the "decision" and "action" easy to state honestly. I'm keeping this
provisional — if the archetype clusters (Lane 3) or CTR-gap (Lane 4) angle turns out to be
richer once I dig into the warehouse release in Week 3+, I'll switch.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

**Question:** Which pages should a content editor review first this week — for refresh,
expansion, protection, pruning, or monitoring?

**Unit of analysis:** one content item (one row = one pseudonymized page, `content_id`), trailing
90-day metrics.

**Decision it improves:** which page(s) get a scarce editor's attention this cycle, out of
thousands of pages, when only a handful of review hours exist.

**Who acts, and what they do:** a content/SEO editor (or their team lead) works down a ranked
queue and, for each flagged page, does one of: refresh the content, expand thin sections,
protect a page that's doing well from being deprioritized, prune a page that's not worth the
upkeep, or just keep monitoring it. The output isn't a probability — it's a short list with a
reason attached, something a person can act on in the next sprint.

**Cost of a wrong call:**
- *False positive* (flagged as "refresh first" but it wasn't worth it): a few editor-hours spent
  on a page that wouldn't have moved the needle — wasted labor, opportunity cost against a page
  that actually needed it.
- *False negative* (a genuinely slipping, high-value page never surfaces): a client-visible page
  keeps losing clicks/visibility for another review cycle before anyone notices — lost traffic
  and, at scale, a credibility problem for the recommendation system itself.
- Because both errors have real but different costs, precision *and* recall matter, and the
  ranked queue needs to surface *why* a page scored the way it did (a reason code), not just a
  bare number — so the editor can sanity-check it before spending time.

**Why data or ML at all (not just a dashboard or an if-statement)?** A single rule like "flag
anything not updated in 90 days" ignores whether the page has any demand, whether it's visible at
all, or whether it's already declining. The real signal is the *interaction* of several things —
age, freshness, visibility, position, trend, volume — which is exactly the kind of tangled,
many-signal pattern a plain if-statement handles poorly but a scored/ranked model (starting from
a transparent baseline) can combine sensibly.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [4]:
!git clone https://github.com/mohamedkhaled600/Starter-Notebook.git
%cd Starter-Notebook/work/notebooks

Cloning into 'Starter-Notebook'...
remote: Enumerating objects: 103, done.
remote: Counting objects: 100% (103/103), done.
remote: Compressing objects: 100% (74/74), done.
remote: Total 103 (delta 22), reused 80 (delta 13), pack-reused 0 (from 0)
Receiving objects: 100% (103/103), 1.83 MiB | 6.26 MiB/s, done.
Resolving deltas: 100% (22/22), done.
/content/Starter-Notebook/work/notebooks/Starter-Notebook/work/notebooks


In [5]:
import pandas as pd

df = pd.read_csv("../../data/raw/content_refresh_anonymized.csv")

n_rows, n_cols = df.shape
n_clients = df["client_id"].nunique()
pct_down = 100 * (df["trend_direction"] == "down").mean()
pct_striking = 100 * (df["position_tier"] == "striking").mean()
pct_visible = 100 * (df["avg_position"] > 0).mean()

print(f"Rows x cols: {n_rows} x {n_cols}, across {n_clients} clients")
print(f"Share of pages trending down (trend_direction == 'down'): {pct_down:.1f}%")
print(f"Share of pages in 'striking distance' position tier: {pct_striking:.1f}%")
print(f"Share of pages that are visible at all (avg_position > 0): {pct_visible:.1f}%")

Rows x cols: 30000 x 44, across 32 clients
Share of pages trending down (trend_direction == 'down'): 54.2%
Share of pages in 'striking distance' position tier: 24.3%
Share of pages that are visible at all (avg_position > 0): 96.0%


**Reading these numbers:** with 30,000 pages across 32 clients, over half (~ 54%) are trending
down and roughly a quarter sit in striking distance of page one — a large, overlapping pool where
prioritization genuinely matters, and where a small ranking mistake (working the wrong page first)
has a real opportunity cost given limited editor time. Almost all pages (~ 96%) are visible at all
(`avg_position > 0`), so this isn't a "mostly invisible" dataset — it's a "mostly visible, often
slipping" one, which is the right shape for a refresh-priority lane rather than, say, a pure
discoverability/indexing lane.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

**What this work can say:**
- *Observed*: on this 90-day window, ~54% of pages show a declining trend (`trend_direction == 'down'`) and
  ~24% sit in striking distance of page one — these are measured facts about the sample, not predictions.
- *Directional / decision-support*: a ranked queue that says "review these pages first, and here's
  the reason code" — a prioritization aid, not a guaranteed-correct verdict.
- *Model-relative*: "the model ranks review-worthy pages better than a simple rule, on held-out
  clients" — a comparison, not an absolute truth claim.

**What this work will never say:**
- Not causal proof — I did not run an experiment, so I can't say refreshing a page *caused* a
  recovery, only that refreshed/high-scoring pages *tend to* look a certain way.
- Not "predicting Google's algorithm" — I'm scoring observable content and search signals, not
  reverse-engineering ranking factors.
- Not a guarantee for any single page — a high score means "worth a look," not "will definitely
  improve if touched."
- No client names, URLs, or private data appear anywhere in this notebook or its outputs.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.